deps.ffmpeg/40-aom.ps1

```diff
@@ -0,0 +1,102 @@
+param(
+    [string] $Name = 'aom',
+    [string] $Version = '3.6.1',
+    [string] $Uri = 'https://aomedia.googlesource.com/aom.git',
+    [string] $Hash = '7ade96172b95adc91a5d85bf80c90989cd543ee8',
+    [array] $FixupPatches = @(
+        @{
+            PatchFile = "${PSScriptRoot}/patches/aom/0001-windows-pkg-config-fix.patch"
+            HashSum = "22f38b49d6307c2ee860b08df7495b5f8894658b451c020f8a13162fd7dd29f4"
+        }
+    ),
+    [array] $Targets = @('x64')
+)
+
+function Setup {
+    Setup-Dependency -Uri $Uri -Hash $Hash -DestinationPath $Path
+
+    if ( ! ( $SkipAll -or $SkipDeps ) ) {
+        Invoke-External pacman.exe -S --noconfirm --needed --noprogressbar yasm
+    }
+}
+
+function Clean {
+    Set-Location $Path
+    if ( Test-Path "build_${Target}" ) {
+        Log-Information "Clean build directory (${Target})"
+        Remove-Item -Path "build_${Target}" -Recurse -Force
+    }
+}
+
+function Configure {
+    Log-Information "Configure (${Target})"
+    Set-Location $Path
+
+    $OnOff = @('OFF', 'ON')
+    $TargetCPUs = @{
+        x64 = 'x86_64'
+    }
+
+    $Options = @(
+        $CmakeOptions
+        "-DBUILD_SHARED_LIBS:BOOL=$($OnOff[$script:Shared.isPresent])"
+        '-DENABLE_DOCS:BOOL=OFF'
+        '-DENABLE_EXAMPLES:BOOL=OFF'
+        '-DENABLE_TESTDATA:BOOL=OFF'
+        '-DENABLE_TESTS:BOOL=OFF'
+        '-DENABLE_TOOLS:BOOL=OFF'
+        "-DAOM_TARGET_CPU=$($TargetCPUs[$Target])"
```

Comment:
It's not because `CMAKE_SYSTEM_PROCESSOR` will not be aware of the target build architecture, it would always default to the host architecture.

Google also doesn't ship toolchain files for MSVC, so you need to enforce this manually.

See also: https://gitlab.kitware.com/cmake/cmake/-/issues/15170

Reply:
Unfortunate, but I understand now.